### PR TITLE
Use week decay

### PR DESCRIPTION
**Pull request by pandadefi**
## Description

Using weekly decay, simplifies integration with other protocols.

## Checklist

- [ ] I have run vyper and solidity linting
- [ ] I have run the tests on my machine
- [ ] I have followed commitlint guidelines
- [ ] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

File: contracts/VotingYFI.vy [hunk 1]
```diff
@@ -175,21 +185,22 @@ def _checkpoint_user(user: address, old_lock: LockedBalance, new_lock: LockedBal
...
     initial_last_point: Point = last_point
     block_slope: uint256 = 0  # dblock/dt
-    if block.timestamp > last_point.ts:
-        block_slope = SCALE * (block.number - last_point.blk) / (block.timestamp - last_point.ts)
+    if self.begining_of_week() > last_point.ts:
+        block_slope = SCALE * (block.number - last_point.blk) / (self.begining_of_week() - last_point.ts)
     
     # apply weekly slope changes and record weekly global snapshots
     t_i: uint256 = self.round_to_week(last_checkpoint)
+    begining_of_week: uint256 = self.begining_of_week()
```
Comment (wavey0x) [inline contracts/VotingYFI.vy:201]: Can you declare this a few lines higher so that it can be used on lines 196 and 197?

**Comment by storming0x** [comment]
We should add a test to surface the bug and validate the change fixes the reported issue

**Comment by pandadefi** [comment]
> We should add a test to surface the bug and validate the change fixes the reported issue

Can you please add the test, that's a good practice to get more familiar with the code and the issue.

**Comment by pandadefi** [comment]
> We should add a test to surface the bug and validate the change fixes the reported issue

Looking a second time at this, the issue is already fixed for `statemind: veYFI balance might not decrease` on https://github.com/yearn/veYFI/pull/179 with MAX_LOCK_DURATION being rounded down to exact weeks and lock being rounded down to weeks on modify lock.

**Comment by pandadefi** [comment]
With the changes, it become possible to increase the power for a past snapshot (within the same week) I don't think that's an issue but we must remember that

**Pull request closed without merging**